app/src/main/java/eu/kanade/tachiyomi/source/LocalSource.kt

```diff
@@ -154,9 +161,114 @@ class LocalSource(
                 }
             }
 
+        // get Metadata from ComicInfo.xml Files
+        val xmlFile = getMangaDirsFiles(manga.url, baseDirsFile)
+            .firstOrNull { it.extension.equals("xml", ignoreCase = true) }
+
+        val noXmlFile = getMangaDirsFiles(manga.url, baseDirsFile)
+            .firstOrNull { it.extension.equals("noxml", ignoreCase = true) }
```

Comment:
How about searching for both files at the same time? Then instead of checking for null in the if statement below check what the extension is as it will either be `xml` or `noxml`

```suggestion
        val metadataRegex = "(no)?xml".toRegex()
        val metadataFile = getMangaDirsFiles(manga.url, baseDirsFile)
            .firstOrNull { it.extension.matches(metadataRegex) } ?: return manga
```

Reply:
Will do thank you

Reply:
I had to change this code a little a while back since the way it was implemented here caused it to exit the function if **no** `.noxml` file was found instead of exiting if it finds a `.noxml` file. 

I now noticed that by continuing to use the Regex `metadataFile` will always be the `.noxml` file which in turn means that if there is both a `.noxml` and a `ComicInfo.xml` file at the chapter root it won't read the `ComicInfo`. 

Now my Question should we keep it that way? 

On the plus side users could create a .noxml file to stop Tachiyomi from parsing ComicInfo files which could maybe save some resources when opening Local Source. But some users may get confused why their ComicInfo files aren't parsed especially since we create .noxml files automatically.

Reply:
Could perhaps use `lastOrNull` if they always come in the order of [`noxml`, `xml`]

Reply:
yeah that sounds good